src/js/models/v2/aboutTheData/BaseReportingPeriodRow.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * BaseReportingPeriodRow.js
+ * Created by Lizzie Salita 12/8/20
+ */
+
+import { formatMoneyWithPrecision, formatNumber, calculatePercentage } from 'helpers/moneyFormatter';
+import { dateFormattedMonthDayYear, showQuarterText } from 'helpers/aboutTheDataHelper';
+
+const BaseReportingPeriodRow = {
+    populate(data, federalBudget) {
+        this._fiscalYear = data.fiscal_year || 0;
+        this._fiscalPeriod = parseInt(data.fiscal_period, 10) || 0;
+        this._budgetAuthority = data.current_total_budget_authority_amount || 0;
+        this._mostRecentUpdate = data.recent_publication_date || null;
+        // eslint-disable-next-line camelcase
+        this._missingTAS = data.tas_account_discrepancies_totals?.missing_tas_accounts_count || 0;
+        this._obligationDifference = data.obligation_difference || 0;
+        this._federalBudget = federalBudget;
+    },
+    get reportingPeriod() {
+        // Add a leading a zero to single-digit periods
+        const paddedPeriod = `${this._fiscalPeriod}`.padStart(2, '0');
+        let period = `P${paddedPeriod}`;
+        if (showQuarterText(this._fiscalPeriod)) {
+            // Indicate the fiscal quarter for periods 3, 6, 9, and 12
+            period = `Q${this._fiscalPeriod / 3} / P${paddedPeriod}`;
+        }
+        else if (this._fiscalPeriod === 2) {
+            // P01 isn't reported, so P02 is cumulative P01 - P02
+            period = 'P01 - P02';
+        }
+        return `FY ${this._fiscalYear}: ${period}`;
+    },
+    get percentOfBudget() {
+        if (this._federalBudget) {
```

Comment:
calculatePercentage should handle bad values for both bad numerators and denominators. The default return value is `--`.